sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -906,20 +910,30 @@ static int sof_card_dai_links_create(struct device *dev,
 
 		ssp_components->name = codec_name;
 		ssp_components->dai_name = info->dai_name;
-		init_dai_link(links, id++, name, info->direction[0],
-			      info->direction[1], cpus, cpu_name,
+		cpus[cpus_id].dai_name = cpu_name;
+		init_dai_link(links + be_id, name, info->direction[0],
+			      info->direction[1], cpus + cpus_id, 1,
 			      ssp_components, 1, dai_id++, NULL, info->ops);
 
-		info->init(NULL, links + id - 1, info, 0);
+		info->init(NULL, links + be_id, info, 0);
+		cpus_id++;
+		be_id++;
```

Comment:
I think you could modify the initi_dai_link() call with links + be_id++......cpus + cpu_id++ too and avoid these statements to increment after every init_dai_link() call?

Reply:
yes, I did it at first, but later I though it may be unacceptable to reviewer. Now I can try it again.